new-log-viewer/src/utils/data.ts

```diff
@@ -38,7 +38,45 @@ const getMapValueWithNearestLessThanOrEqualKey = <T>(
         map.get(lowerBoundKey) as T;
 };
 
+/**
+ * Creates an array of numbers in a specified range [startNum, startNum + steps].
+ *
+ * @param start The value of the start parameter (or `0` if the parameter was not supplied).
+ * @param stop The value of the stop parameter.
+ * @param step The value of the step parameter (or `1` if the parameter was not supplied).
+ * - For a positive step, the contents of a range r are determined by the formula `r[i] = start +
+ * step*i` where `i >= 0` and `r[i] < stop`.
+ * - For a negative step, the contents of the range are still determined by the formula `r[i] =
+ * start + step*i`, but the constraints are `i >= 0` and `r[i] > stop`.
+ * @return An array of numbers from `start` to `stop` (exclusive) with a step of `step`.
+ * @throws {Error} if `step` is 0.
+ */
+const range = (start: number, stop: Nullable<number> = null, step: number = 1): number[] => {
+    const result: number[] = [];
+
+    if (0 === step) {
+        throw new Error("Step cannot be zero.");
+    }
+    if (null === stop) {
+        stop = start;
+        start = 0;
```

Comment:
```suggestion
```

I don't think we need this, do we?

Reply:
If the user only provides a single argument which is to indicate the ending index, it's assumed that the starting index is 0. I'm not sure if there's a better way to name the parameters though.

Let me document the behaviour first.

Reply:
Oh, oh, I see. Could we maybe do something like (this is just the idea, not the exact syntax, lol) `range(args: number | {start: number, stop: number, step?: number})`?